src/test/java/de/mediathekview/mserver/crawler/zdf/ZdfVideoUrlOptimizerTest.java

```diff
@@ -0,0 +1,88 @@
+package de.mediathekview.mserver.crawler.zdf;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import de.mediathekview.mserver.testhelper.WireMockTestBase;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ZdfVideoUrlOptimizerTest extends WireMockTestBase {
+
+  private ZdfVideoUrlOptimizer target;
+
+  @Before
+  public void setup() {
+    target = new ZdfVideoUrlOptimizer();
+  }
+
+  @Test
+  public void getOptimizedUrlNormalTestAlreadyBestQuality() {
+    final String url = WireMockTestBase.MOCK_URL_BASE + "/video_2328k_p35v11.mp4";
+
+    assertGetOptimizedUrlNormal(url, url);
+  }
+
+  @Test
+  public void getOptimizedUrlNormalTestBetterUrlExists() {
+    final String url = WireMockTestBase.MOCK_URL_BASE + "/video_1456k_p13v11.mp4";
+    final String expectedUrl = WireMockTestBase.MOCK_URL_BASE + "/video_2328k_p35v11.mp4";
+
+    setupHeadResponse("/video_2328k_p35v11.mp4", 200);
+    assertGetOptimizedUrlNormal(expectedUrl, url);
+  }
+
+  @Test
+  public void getOptimizedUrlNormalTestBetterUrlNotExists() {
+    final String url = WireMockTestBase.MOCK_URL_BASE + "/video_2256k_p14v12.mp4";
+
+    setupHeadResponse("/video_2328k_p35v12.mp4", 404);
+    assertGetOptimizedUrlNormal(url, url);
+  }
+
+  @Test
+  public void determineUrlHdTestFirstUrlExists() {
+    final String url = WireMockTestBase.MOCK_URL_BASE + "/video_1456k_p13v12.mp4";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "/video_1456k_p13v12.mp4" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)